File: 06_mar/LL_AddFront/helpers/SList.cpp
#include <iostream>
#include "SList.h"

SList::SList()
{
	head = NULL;
}

void SList::addFront(int val)
{
	node * temp = new node;
	temp -> value = val;
	if (head == NULL) temp -> next = NULL;
	else temp -> next = head;

	head = temp;
}

node * SList::removeFront()
{
	node * temp = head;
	head = temp -> next;
	temp -> next = NULL;
	return temp;
}

void SList::printList()
{
	node * curr = head;

	while (curr)
	{
		std::cout << curr -> value << std::endl;

		curr = curr -> next;
	}
}

SList::~SList()
{
	node * p = head;
	node * q = head;
	while(q)
	{
		p = q;
		q = p -> next;
		if (q) delete p;
	}
}

File: js/07_mar/LL_07_mar/helpers/SList.java
package helpers;

import static java.lang.System.out;

public class SList {
	Node head;

	public SList() {
		this.head = null;
	}

	public void addFront(int val) {
		Node tempNode = new Node(val);
		tempNode.next = this.head;
		this.head = tempNode;
	}

	public Node removeFront() {
		Node tempNode = this.head;
		this.head = tempNode.next;
		tempNode.next = null;
		return tempNode;
	}

	public boolean contains(int val) {
		Node current = this.head;

		while (current != null) {
			if (current.value == val) {
				return true;
			}

			current = current.next;
		}

		return false;
	}

	public Node front() {
		return this.head;
	}

	public void printList() {
		Node current = this.head;

		while (current != null) {
			out.println(current.value);
			current = current.next;
		}
	}
}
